core/validation/failures.py

```diff
@@ -85,6 +85,23 @@ def __str__(self):
         return f'Non Unique Failure: Sheet "{self.sheet}" column "{self.column}" should ' f"contain only unique values."
 
 
+@dataclass
+class NonUniqueCompositeKeyFailure(ValidationFailure):
+    """Class representing a non-unique-composite_key failure."""
+
+    sheet: str
+    cols: tuple
+    index: int
+
+    def __str__(self):
+        """Method to get the string representation of the non-unique-composite_key failure."""
+        cols_str = ", ".join(str(i) for i in self.cols)
+        return (
+            f'Non Unique Row Failure: Sheet "{self.sheet}"; '
+            f'Columns "{cols_str}" contain non-unique values at row "{self.index}"'
```

Comment:
maybe this could be a bit clearer? ie something like:
"Columns "{cols_str}" contain non-unique combination of values at row "{self.index}"